### PR TITLE
migrating websocketpp/0.8.1@bincrafters/stable to CCI

### LINKED ISSUE #1156: [request] websocketpp/0.8.1
### Package Details
  * Package Name/Version: **websocketpp/0.8.1**
  * Website: **http://www.zaphoyd.com/websocketpp**
  * Source code: **https://github.com/zaphoyd/websocketpp/tree/develop**

### Description Of The Library / Tool
C++ webscoket library on top of boost asio

Currently this is hosted by @bincrafters-user but I would like to migrate it to CCI.
https://github.com/bincrafters/conan-websocketpp

@Croydon would this be allowed? I would like to confirm before doing the work =)

### PR DESCRIPTION
**Pull request by prince-chrismc**
Based on: https://github.com/bincrafters/conan-websocketpp/tree/fea456bee75c7e8459e358207e2477c8243ef735

fixes #1156

Specify library name and version:  **websocketpp/0.8.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Few difference from bincrafters recipe:
- update deps version
- moved patches to `conandata.yml`
- removed build step

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Hmm I did not get those errors with the hooks on Windows =?

**Comment by uilianries** [comment]
The CPP_11 is enabled by default, but the defines declared in the cmakefile are not exported in the Conan recipe: https://github.com/zaphoyd/websocketpp/blob/0.8.1/CMakeLists.txt#L102

**Comment by prince-chrismc** [comment]
They are controlled automatically by the headers themselves

https://github.com/zaphoyd/websocketpp/search?q=_WEBSOCKETPP_CPP11_INTERNAL_&type=

In fact some are missing from the CMakeLists.txt =?

File: recipes/websocketpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    generators = ["cmake"]
+    options = {'asio': ['boost', 'standalone']}
+    default_options = {'asio': 'boost'}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires.add('openssl/1.1.1d')
```
Comment (SSE4) [inline recipes/websocketpp/all/conanfile.py]:
```suggestion
        self.requires.add('openssl/1.1.1e')
```

**Pull request merged**